queries/cpp/highlights.scm

```diff
@@ -71,32 +71,13 @@
 "operator" @function
 "static_assert" @function.builtin
 
-(call_expression
-  function: (qualified_identifier
-              name: (identifier) @function.call))
-(call_expression
-  function: (qualified_identifier
-              name: (qualified_identifier
-                      name: (identifier) @function.call)))
-(call_expression
-  function:
-      (qualified_identifier
-        name: (qualified_identifier
-              name: (qualified_identifier
-                      name: (identifier) @function.call))))
-(call_expression
-  function: (template_function
-              name: (identifier) @function.call))
-(call_expression
-  function: (qualified_identifier
-              name: (template_function
-                      name: (identifier) @function.call)))
-(call_expression
-  function:
-      (qualified_identifier
-        name: (qualified_identifier
-              name: (template_function
-                      name: (identifier) @function.call))))
+((qualified_identifier
+     (identifier) @function.call) @_parent
+ (#has-ancestor? @_parent call_expression))
```

Comment:
Oh, sorry I forgot about this issue. Maybe we can revert that commit?

I would avoid has-ancestor. Getting the parent of a node is very costly and it will run on almost every node. The hardcoded pattern is definitely better. Maybe we can run has-ancestor only at the third or fourth level which will definitely be rarer

Reply:
Sounds good. Updated.

Reply:
Also refactored the query file a bit to group related queries together, and cleaned up a bit of the indenting and labelling of nodes. If you prefer the old way (explicit labels), I can go back, but I don't think they're terribly useful for cpp.